src/grandorgue/sound/GOSoundStream.cpp

```diff
@@ -76,33 +130,65 @@ GOSoundStream::DecodeBlockFunction GOSoundStream::getDecodeBlockFunction(
   GOSoundResample::InterpolationType interpolation,
   bool is_end) {
   if (compressed && !is_end) {
-    /* TODO: Add support for polyphase compressed decoders. Fallback to
-     * linear interpolation for now. */
-    if (channels == 1) {
-      if (bits_per_sample >= 20)
+    if (interpolation == GOSoundResample::GO_POLYPHASE_INTERPOLATION) {
+      if (channels == 1) {
+        if (bits_per_sample >= 20)
+          return &GOSoundStream::DecodeBlock<
+            GOSoundResample::PolyphaseResampler,
+            StreamCacheReadAheadWindow<
+              true,
+              GOSoundResample::PolyphaseResampler::VECTOR_LENGTH,
+              1>>;
+
+        assert(bits_per_sample >= 12);
         return &GOSoundStream::DecodeBlock<
-          GOSoundResample::LinearResampler,
-          StreamCacheWindow<true, 1>>;
-
-      assert(bits_per_sample >= 12);
-      return &GOSoundStream::DecodeBlock<
-        GOSoundResample::LinearResampler,
-        StreamCacheWindow<false, 1>>;
-    } else if (channels == 2) {
-      if (bits_per_sample >= 20)
+          GOSoundResample::PolyphaseResampler,
+          StreamCacheReadAheadWindow<
+            false,
+            GOSoundResample::PolyphaseResampler::VECTOR_LENGTH,
+            1>>;
+      } else if (channels == 2) {
+        if (bits_per_sample >= 20)
+          return &GOSoundStream::DecodeBlock<
+            GOSoundResample::PolyphaseResampler,
+            StreamCacheReadAheadWindow<
+              true,
+              GOSoundResample::PolyphaseResampler::VECTOR_LENGTH,
+              2>>;
+
+        assert(bits_per_sample >= 12);
+        return &GOSoundStream::DecodeBlock<
+          GOSoundResample::PolyphaseResampler,
+          StreamCacheReadAheadWindow<
+            false,
+            GOSoundResample::PolyphaseResampler::VECTOR_LENGTH,
+            2>>;
+      }
+    } else {
+      if (channels == 1) {
+        if (bits_per_sample >= 20)
+          return &GOSoundStream::DecodeBlock<
+            GOSoundResample::LinearResampler,
+            StreamCacheWindow<true, 1>>;
+
+        assert(bits_per_sample >= 12);
```

Comment:
What does assert() in case of failure (I've read a program Abort) ? Maybe something more user-friendly should be good no ?

Reply:
It checks the condition in debug mode and generates a signal to debugger if it is not satisfied.

It does nothing in a normal running mode.

Usually if checks the correctness of the program so it shouldn't fail.

This line is a copy of line 159 that was before this PR.